shell/platform/darwin/ios/framework/Source/FlutterUndoManagerPlugin.mm

```diff
@@ -49,45 +45,47 @@ - (void)handleMethodCall:(FlutterMethodCall*)call result:(FlutterResult)result {
 }
 
 - (NSUndoManager*)undoManager {
-  return _viewController.undoManager;
+  return self.undoManagerDelegate.viewController.undoManager;
 }
 
 - (void)resetUndoManager API_AVAILABLE(ios(9.0)) {
   [[self undoManager] removeAllActionsWithTarget:self];
 }
 
 - (void)registerUndoWithDirection:(FlutterUndoRedoDirection)direction API_AVAILABLE(ios(9.0)) {
-  [[self undoManager] beginUndoGrouping];
-  [[self undoManager] registerUndoWithTarget:self
-                                     handler:^(id target) {
-                                       // Register undo with opposite direction.
-                                       FlutterUndoRedoDirection newDirection =
-                                           (direction == FlutterUndoRedoDirectionRedo)
-                                               ? FlutterUndoRedoDirectionUndo
-                                               : FlutterUndoRedoDirectionRedo;
-                                       [target registerUndoWithDirection:newDirection];
-                                       // Invoke method on delegate.
-                                       [_undoManagerDelegate flutterUndoManagerPlugin:self
-                                                              handleUndoWithDirection:direction];
-                                     }];
-  [[self undoManager] endUndoGrouping];
+  NSUndoManager* undoManager = [self undoManager];
+  [undoManager beginUndoGrouping];
+  [undoManager registerUndoWithTarget:self
+                              handler:^(FlutterUndoManagerPlugin* target) {
+                                // Register undo with opposite direction.
+                                FlutterUndoRedoDirection newDirection =
+                                    (direction == FlutterUndoRedoDirectionRedo)
+                                        ? FlutterUndoRedoDirectionUndo
+                                        : FlutterUndoRedoDirectionRedo;
+                                [target registerUndoWithDirection:newDirection];
+                                // Invoke method on delegate.
+                                [target.undoManagerDelegate flutterUndoManagerPlugin:target
```

Comment:
Changed `_undoManagerDelegate` to `target.undoManagerDelegate` where and passed `target` into `flutterUndoManagerPlugin` so `self` isn't retained in this block.